File: Android/SayHi/app/src/main/java/de/oklemenz/sayhi/model/StageTag.java
package de.oklemenz.sayhi.model;

import java.io.Serializable;

/**
 * Created by Oliver Klemenz on 22.02.17.
 */

public class StageTag implements Serializable {

    public String key = "";
    public String langCode = "";
    public String name = "";
    public String categoryKey = "";
    public String categoryName = "";
    public String primaryLangKey = "";
    public Long counter = 1L;

    public StageTag(String key, String langCode, String name, String categoryKey, String categoryName, String primaryLangKey, Long counter) {
        this.key = key;
        this.langCode = langCode;
        this.name = name;
        this.categoryKey = categoryKey;
        this.categoryName = categoryName;
        this.primaryLangKey = primaryLangKey;
        this.counter = counter;
    }

    public Tag tag() {
        Tag tag = new Tag(
                key,
                langCode,
                name,
                categoryKey,
                primaryLangKey);
        tag.stage = true;
        return tag;
    }
}

File: iOS/SayHi/SayHi/StageTag.swift
//
//  StageTag.swift
//  SayHi
//
//  Created by Klemenz, Oliver on 23.12.16.
//  Copyright © 2016 Klemenz, Oliver. All rights reserved.
//

import Foundation

struct StageTag {
    var key: String = ""
    var langCode: String = ""
    var name: String = ""
    var categoryKey: String = ""
    var categoryName: String = ""
    var primaryLangKey: String = ""
    var counter: Int = 1
    
    var tag: Tag {
        let tag = Tag(key: key,
                      langCode: langCode,
                      name: name,
                      categoryKey: categoryKey,
                      primaryLangKey : primaryLangKey)
        tag.stage = true
        return tag
    }
}
